lms/djangoapps/bookmarks/tests/test_api.py

```diff
@@ -0,0 +1,175 @@
+        """
+        bookmark_data = api.get_bookmark(user=self.user, usage_key=self.vertical.location)
+        self.assert_bookmark_response(bookmark_data, self.bookmark)
+
+        # With Optional fields.
+        bookmark_data = api.get_bookmark(
+            user=self.user,
+            usage_key=self.vertical.location,
+            fields=self.all_fields
+        )
+        self.assert_bookmark_response(bookmark_data, self.bookmark, optional_fields=True)
+
+    def test_get_bookmark_raises_error(self):
+        """
+        Verifies that get_bookmark raises error as expected.
+        """
+        with self.assertRaises(ObjectDoesNotExist):
+            api.get_bookmark(user=self.other_user, usage_key=self.vertical.location)
+
+    def test_get_bookmarks(self):
+        """
+        Verifies that get_bookmarks returns data as expected.
+        """
+        # Without course key.
+        bookmarks_data = api.get_bookmarks(user=self.user)
+        self.assertEqual(len(bookmarks_data), 2)
+        # Assert them in ordered manner.
+        self.assert_bookmark_response(bookmarks_data[0], self.bookmark_2)
+        self.assert_bookmark_response(bookmarks_data[1], self.bookmark)
+
+        # With course key.
+        bookmarks_data = api.get_bookmarks(user=self.user, course_key=self.course.id)
+        self.assertEqual(len(bookmarks_data), 1)
+        self.assert_bookmark_response(bookmarks_data[0], self.bookmark)
+
+        # With optional fields.
+        bookmarks_data = api.get_bookmarks(user=self.user, course_key=self.course.id, fields=self.all_fields)
+        self.assertEqual(len(bookmarks_data), 1)
+        self.assert_bookmark_response(bookmarks_data[0], self.bookmark, optional_fields=True)
+
+        # Without Serialized.
+        bookmarks = api.get_bookmarks(user=self.user, course_key=self.course.id, serialized=False)
+        self.assertEqual(len(bookmarks), 1)
+        self.assertEqual(bookmarks[0], self.bookmark)
+
+    def test_create_bookmark(self):
+        """
+        Verifies that create_bookmark create & returns data as expected.
+        """
+        self.assertEqual(len(api.get_bookmarks(user=self.user, course_key=self.course.id)), 1)
+
+        api.create_bookmark(user=self.user, usage_key=self.vertical_1.location)
+
+        self.assertEqual(len(api.get_bookmarks(user=self.user, course_key=self.course.id)), 2)
+
+    def test_create_bookmark_do_not_create_duplicates(self):
+        """
+        Verifies that create_bookmark do not create duplicate bookmarks.
+        """
+        self.assertEqual(len(api.get_bookmarks(user=self.user, course_key=self.course.id)), 1)
+        api.create_bookmark(user=self.user, usage_key=self.vertical_1.location)
+
+        self.assertEqual(len(api.get_bookmarks(user=self.user, course_key=self.course.id)), 2)
+
+        api.create_bookmark(user=self.user, usage_key=self.vertical_1.location)
+        self.assertEqual(len(api.get_bookmarks(user=self.user, course_key=self.course.id)), 2)
```

Comment:
You can also check that the data returned by `create_bookmark` is the same for both calls.